packages/api/src/controllers/stream.js

```diff
@@ -53,6 +54,51 @@ const hackMistSettings = (req, profiles) => {
   });
 };
 
+async function validatePushTarget(userId, profileNames, pushTargetRef) {
+  const { profile, id, spec } = pushTargetRef;
+  if (!profileNames.has(profile) && profile !== "source") {
+    throw new BadRequestError(
+      `push target must reference existing profile. not found: "${profile}"`
+    );
+  }
+  if (!!spec === !!id) {
+    throw new BadRequestError(
+      `push target must have either an "id" or a "spec"`
+    );
+  }
+  if (id) {
+    if (!(await db.pushTarget.hasAccess(id, userId))) {
+      throw new BadRequestError(`push target not found: "${id}"`);
```

Comment:
Seems to me that this should be either a 404 "Push Target Not Found" or a 403 "You May Not Add This Push Target" kind of thing. If they already have the id of someone else's push target, we probably don't accomplish too much by hiding it with a 404, so I'm thinking the 403 might be appropriate.

Reply:
I guess there are 2 subjects here: 
 1. If we should separate the error cases when the stream doesn't exist from when it is from another user;
 2. On any case, what status code to use.

I'm ok with making this return a "real" `404` when the push target doesn't exist and `403` when it exists but is from another user, but I'll explain my reasoning below and you let me know if you do prefer it that way.

### 1
I think it ends up a [simpler code](https://github.com/livepeer/livepeer-com/blob/b8ac0cd01b7620bae5c94abe9b8a4ee5a0b79874/packages/api/src/controllers/stream.js#L956) overall to treat both not existence and lack of permission the same way, so that we don't need separate `if` checks in the code which becomes more verbose. That `hasAccess` (and the underlying `getAuthed`) function checks both of those and returns `true` iff push target exists and user has access. 

So maybe the question should be the opposite, of whether it's worth to break up the conditions to be explicit about whether the object exists and is forbidden or it just doesn't exist. IMO there's a small gain in hiding the existence fact, then since the code felt a bit cleaner to me I thought it was worth adotping the pattern of "404"s when trying to access a cross-user resource.

### 2
I thought it could be misleading to return a `404` on a `/stream/*` endpoint, when actually the stream in question does exist but something in the payload that had a bad reference, either of a push target that doesn't exist or a push target from another user. 

Especially on the `PATCH` scenario, a `404` from a request to `/stream/abc-1234-def` request would make me think that the stream in question doesn't exist instead. So then this is not necessarily an HTTP `404` error, referring to the resource being accessed*, but a problem with the payload, which is how I ended up on the `400`.

\* This may be just my understanding of the 404 though, that it should refer to the resource in the URL. Couldn't find a definite answer with a quick search.

Reply:
Ah sure, I didn't wrap my head around the fact this was in `/stream` rather than `/push-target` - sounds fine to me